src/sql/azdata.d.ts

```diff
@@ -2642,24 +2642,78 @@ declare module 'azdata' {
 	 * Represents a provider of resource
 	 */
 	export interface ResourceProvider {
+		/**
+		 * Creates a firewall rule for the given account
+		 * @param account Account with which firewall rule request will be made.
+		 * @param firewallruleInfo Firewall rule creation information
+		 */
 		createFirewallRule(account: Account, firewallruleInfo: FirewallRuleInfo): Thenable<CreateFirewallRuleResponse>;
+
+		/**
+		 * Handles the response from the firewall rule creation request
+		 * @param errorCode Error code from the firewall rule creation request
+		 * @param errorMessage Error message from the firewall rule creation request
+		 * @param connectionTypeId Connection type id of the firewall rule creation request
+		 */
 		handleFirewallRule(errorCode: number, errorMessage: string, connectionTypeId: string): Thenable<HandleFirewallRuleResponse>;
 	}
 
+	/**
+	 * Firewall rule creation information
+	 */
 	export interface FirewallRuleInfo {
+		/**
+		 * Start of the IP address range
+		 */
 		startIpAddress?: string | undefined;
+		/**
+		 * End of the IP address range
+		 */
 		endIpAddress?: string | undefined;
+		/**
+		 * Fully qualified name of the server to create a new firewall rule on
+		 */
 		serverName: string;
+		/**
+		 * Firewall rule name to set
+		 */
+		firewallRuleName: string;
```

Comment:
@cheenamalhotra @alanrenmsft 

Why were these not added to the proposed API file?

We should also be careful of adding non-optional properties to existing types. If an extension was calling this directly then we could break compilation of them by making this change (in addition to then also potentially getting a null value for this until the extension updates). In this case unlikely to be an issue but it's still good to be in the habit of being more deliberate about any potentially breaking changes to the extension API.